File: app/src/main/java/org/tasks/injection/InjectingFragment.java
package org.tasks.injection;

import android.app.Activity;
import androidx.fragment.app.Fragment;

public abstract class InjectingFragment extends Fragment {

  private boolean injected;

  @Override
  public void onAttach(Activity activity) {
    super.onAttach(activity);

    if (!injected) {
      inject(((InjectingActivity) activity).getComponent().plus(new FragmentModule(this)));
      injected = true;
    }
  }

  protected abstract void inject(FragmentComponent component);
}

File: app/src/main/java/org/tasks/injection/InjectingFragment.kt
package org.tasks.injection

import android.app.Activity
import androidx.fragment.app.Fragment

abstract class InjectingFragment : Fragment() {
    private var injected = false
    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        if (!injected) {
            inject((activity as InjectingActivity).component.plus(FragmentModule(this)))
            injected = true
        }
    }

    protected abstract fun inject(component: FragmentComponent)
}